packages/query-core/src/streamedQuery.ts

```diff
@@ -55,19 +82,16 @@ export function streamedQuery<
 
       // don't append to the cache directly when replace-refetching
       if (!isRefetch || refetchMode !== 'replace') {
-        context.client.setQueryData<Array<TQueryFnData>>(
-          context.queryKey,
-          (prev = []) => {
-            return addToEnd(prev, chunk, maxChunks)
-          },
+        context.client.setQueryData<TData>(context.queryKey, (prev) =>
+          reducer(prev ?? initialValue, chunk),
```

Comment:
we need to explicitly check for `undefined` here because it’s the only indicator for “empty” we have. `null` is technically allowed:

```suggestion
          reducer(prev === undefined ? initialValue : prev, chunk),
```